File: app/src/main/java/com/example/cleanarchitecture_toyproject/viewmodel/mapper/UserModelMapper.java
package com.example.cleanarchitecture_toyproject.viewmodel.mapper;

import android.util.Log;
import com.example.cleanarchitecture_toyproject.domain.User;
import com.example.cleanarchitecture_toyproject.viewmodel.model.UserModel;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

//UserModelMapper
public class UserModelMapper {
    public UserModelMapper() {
    }

    private UserModel transform(User user) {
        if (user == null) {
            throw new IllegalArgumentException("cannot transform a null value");
        }
        UserModel userModel = new UserModel(user.getId(), user.getLogin(), user.getAvatar_url(), user.getChecked());

        userModel.setLogin(user.getLogin());
        userModel.setAvatar_url(user.getAvatar_url());
        Log.d("userModeldatamapper 1", userModel.getLogin());
        Log.d("userModeldatamapper 2", userModel.getAvatar_url());
        Log.d("userModeldatamapper 3", String.valueOf(userModel.getChecked()));
        return userModel;
    }

    public List<UserModel> transform(List<User> userCollection) {
        List<UserModel> userModelCollection;

        if (userCollection != null && !userCollection.isEmpty()) {
            userModelCollection = new ArrayList<>();
            for (User user : userCollection) {
                userModelCollection.add(transform(user));
            }
        } else {
            userModelCollection = Collections.emptyList();
        }
        return userModelCollection;
    }

    public User transform(UserModel userModel) {
        return new User(userModel.getId(), userModel.getLogin(), userModel.getAvatar_url(), userModel.getChecked());
    }
}

File: app/src/main/java/com/example/cleanarchitecture_toyproject/presentation/mapper/UserModelMapper.kt
package com.example.cleanarchitecture_toyproject.presentation.mapper

import android.util.Log
import com.example.cleanarchitecture_toyproject.domain.User
import com.example.cleanarchitecture_toyproject.presentation.model.UserModel

import java.util.ArrayList

//UserModelMapper
class UserModelMapper {

    //mapper에는 private function 만들지 않기
/*
    private fun transform(target: User): UserModel = with(target){
        if (target == null) {
            throw IllegalArgumentException("cannot transform a null value")
        }


        Log.d("userModeldatamapper 1", login)
        Log.d("userModeldatamapper 2", avatar_url)
        Log.d("userModeldatamapper 3", checked.toString())
        return UserModel(id, login, avatar_url, checked)
    }*/

    fun transform(target: List<User>): List<UserModel> = with(target){
        return map { UserModel(it.id, it.login, it.avatar_url, it.checked) }
    }

    fun transform(userModel: UserModel): User = with(userModel){
        Log.d("User Transform", userModel.avatar_url.toString())
        return User(id, login, avatar_url, checked)
    }
}
